File: src/services/CompleteRequestService.js
import axios from './AxiosInstance'

const getSummary = () => {
    return axios.get('/complete-request/summary')
}

const listCompleteRequests = (pageNumber, pageSize) => {
    return axios.get('/complete-request?pageNumber=' + pageNumber + '&pageSize=' + pageSize)
        .then(response => {
            response.data.content.forEach(completeRequest => {
                completeRequest.totalJobCount = completeRequest.trackedLists.map(trackedList => trackedList.totalJobCount).reduce((acc, value) => acc + value, 0);
                completeRequest.completedJobCount = completeRequest.trackedLists.map(trackedList => trackedList.completedJobCount).reduce((acc, value) => acc + value, 0);
                completeRequest.errorJobCount = completeRequest.trackedLists.map(trackedList => trackedList.errorJobCount).reduce((acc, value) => acc + value, 0);
                if (Object.keys(completeRequest.results).length === 0)
                    completeRequest.result = "Not Found"
                else if (Object.keys(completeRequest.results).length < completeRequest.targetCount)
                    completeRequest.result = "Partial"
                else
                    completeRequest.result = "Found"
                
                switch(completeRequest.requestType){
                    case "2500":
                        completeRequest.requestType = "WPA";
                        break;
                    case "1000":
                        completeRequest.requestType = "NTLM";
                        break;
                    case "0":
                        completeRequest.requestType = "MD5";
                        break;
                    default:
                        completeRequest.requestType = "Unknown"
                }
            })
            return response;
        })
}

const deleteCompleteRequest = (completeRequestId) => {
    return axios.delete('/complete-request/' + completeRequestId)
}

const CompleteRequestService = {
    getSummary,
    listCompleteRequests,
    deleteCompleteRequest,
}

export default CompleteRequestService;

File: kraken-server/src/main/java/com/arcaneiceman/kraken/krakenserver/service/CompleteRequestService.java
package com.arcaneiceman.kraken.krakenserver.service;

import com.arcaneiceman.kraken.krakenserver.controller.io.CompleteRequestIO;
import com.arcaneiceman.kraken.krakenserver.domain.CompleteRequest;
import com.arcaneiceman.kraken.krakenserver.domain.abs.TrackedList;
import com.arcaneiceman.kraken.krakenserver.domain.embeddedable.CompleteTrackedList;
import com.arcaneiceman.kraken.krakenserver.domain.enumerations.TrackingStatus;
import com.arcaneiceman.kraken.krakenserver.repository.CompleteRequestRepository;
import com.arcaneiceman.kraken.krakenserver.util.exceptions.SystemException;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

import static org.zalando.problem.Status.NOT_FOUND;

@Service
@Transactional
public class CompleteRequestService {

    private CompleteRequestRepository completeRequestRepository;
    private UserService userService;

    public CompleteRequestService(CompleteRequestRepository completeRequestRepository,
                                  UserService userService) {
        this.completeRequestRepository = completeRequestRepository;
        this.userService = userService;
    }

    public CompleteRequest create(String requestType,
                                  String requestName,
                                  String requestMetadata,
                                  Integer targetCount,
                                  List<TrackedList> trackedLists,
                                  String results,
                                  TrackingStatus status) {
        return completeRequestRepository.save(new CompleteRequest(null, requestType, requestName, requestMetadata,
                targetCount, trackedLists.stream().map(CompleteTrackedList::new).collect(Collectors.toList()),
                results, status, userService.getUserOrThrow()));
    }

    public Page<CompleteRequest> list(Pageable pageable) {
        return completeRequestRepository.findByOwner(pageable, userService.getUserOrThrow());
    }

    public CompleteRequest get(String id) {
        return completeRequestRepository.findByIdAndOwner(id, userService.getUserOrThrow())
                .orElseThrow(() -> new SystemException(432, "Complete Request with id " + id + " not found", NOT_FOUND));
    }

    public void delete(String id) {
        CompleteRequest completeRequest = completeRequestRepository.findByIdAndOwner(id, userService.getUserOrThrow())
                .orElseThrow(() -> new SystemException(432, "Complete Request with id " + id + " not found", NOT_FOUND));
        completeRequestRepository.delete(completeRequest);
    }

    public CompleteRequestIO.Summary.Response summary() {
        return new CompleteRequestIO.Summary.Response(
                completeRequestRepository.findByOwner(userService.getUserOrThrow()).size(),
                completeRequestRepository.findByOwnerAndResultsNot(userService.getUserOrThrow(), "{}").size());
    }

}
